pbf/templates/player.html

```diff
@@ -205,13 +205,27 @@ <h4>Impending:</h4>
   <ul>
     <div class="container-fluid">
       <div class="row">
-	{% for card in player.impending.all %}
+	{% for i in impending_with_energy.all %}
```

Comment:
My personal feelings only, and ultimately I leave the decision to the maintainer and the implementor (you):

I feel like it's unnecessary extra work to have to set `impending_with_energy` on everything that renders `player`.

I understand why it's done; you need to reference the `energy` and `in_play` fields of the relationship, not just the card.

I will present an alternative, and leave the decision to you:

use `for card in player.impending_with_energy.all` here.
(edit: I meant `card`, not `i`; just whatever matches the variable name below)

Below:
Every instance of `i.card` just changes to `card`.
Every instance of `i.energy` changes to `card.gameplayerimpendingwithenergy__energy`.
Every instance of `i.in_play` changes to `card.gameplayerimpendingwithenergy__in_play`.

The benefit of doing so is you'll no longer need that `render_player_view` function anymore.

If `gameplayerimpendingwithenergy` is too long (it certainly is a mouthful, I agree), you might look into using https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.ForeignKey.related_query_name to shorten it?

Reply:
Thanks for the ideas! I didn't love this part...

Reply:
I need to make a few edits to my suggestion since it didn't quite work as I intended it to, but I am hoping I can figure something out.

Reply:
Okay, it turns out it needs to be `card.gameplayerimpendingwithenergy_set.first.energy` and `card.gameplayerimpendingwithenergy_set.first.in_play`, according to https://forum.djangoproject.com/t/reading-extra-fields-in-a-manytomany-through-table/10676. I'll edit my original message

Reply:
Turns out `for i in player.gameplayerimpendingwithenergy_set.all` works. It gets you all the relationship objects of the player. So all instances of `i` can just remain as you had them.

Reply:
Thanks! I tested this out with multiple games/spirits and it seemed to fetch the expected dataset :+1: